src/iframe-helper.js

```diff
@@ -380,3 +380,61 @@ function parseIfNeeded(data) {
   }
   return data;
 }
+
+
+
+/**
+ * Manages a postMessage API for an iframe with a subscription message and
+ * a way to broadcast messages to all subscribed windows, which
+ * in turn must all be descendants of the contentWindow of the iframe.
+ */
+export class SubscriptionApi {
```

Comment:
Well, instantiating a class brings extra cost. If the cost pays for nothing, we'd better optimize it.

As you can see, AMP cares about performance so much. In many places, we are trying hard to inline function calls to have some performance gain. For example: #4115